presto-native-execution/presto_cpp/main/operators/tests/BroadcastTest.cpp

```diff
@@ -143,4 +221,125 @@ TEST_F(BroadcastTest, broadcastWrite) {
   runBroadcastTest({data});
 }
 
+TEST_F(BroadcastTest, endToEndWithMultipleWriteNodes) {
+  auto data1 = vectorMaker_.rowVector({
+      makeFlatVector<int32_t>({1, 2, 3, 4, 5, 6}),
+      makeFlatVector<int64_t>({10, 20, 30, 40, 50, 60}),
+  });
+  auto data2 = vectorMaker_.rowVector({
+      makeFlatVector<int32_t>({11, 21, 31, 41, 51, 61}),
+      makeFlatVector<int64_t>({102, 203, 304, 405, 506, 607}),
+  });
+  auto basePath = exec::test::TempDirectoryPath::create()->path;
+
+  // Execute write 1
+  auto [taskCursor1, results1] = executeBroadcastWrite({data1}, basePath);
+  auto broadcastFilePath1 =
+      results1.back()->childAt(0)->as<SimpleVector<StringView>>()->valueAt(0);
+
+  // Execute write 2
+  auto [taskCursor2, results2] = executeBroadcastWrite({data2}, basePath);
+  auto broadcastFilePath2 =
+      results2.back()->childAt(0)->as<SimpleVector<StringView>>()->valueAt(0);
+
+  // Create exchange source with file from write 1 and write 2
+  auto readerPlan =
+      exec::test::PlanBuilder().exchange(asRowType(data1->type())).planNode();
+  exec::test::CursorParameters broadcastReadParams;
+  broadcastReadParams.planNode = readerPlan;
+
+  // Read back result
+  auto [taskCursorReadNode, broadcastReadResults] =
+      readCursor(broadcastReadParams, [&](auto* task) {
+        auto fileInfos = fmt::format(
+            "{}, {}",
+            fmt::format(kBroadcastFileInfoFormat, broadcastFilePath1),
+            fmt::format(kBroadcastFileInfoFormat, broadcastFilePath2));
+
+        auto split = exec::Split(
+            std::make_shared<exec::RemoteConnectorSplit>(fmt::format(
```

Comment:
This code repeats. Consider extracting a helper function to create a list of splits from a list of BroadcastFileInfo.

Reply:
done